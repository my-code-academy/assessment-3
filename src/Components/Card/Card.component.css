.date {
  font-size: 15px;
}

.title {
  font-size: 22px;
}

.abstract {
  font-size: 20px;
  color: #606060;
}

.card {
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  padding: 0px;
  text-align: center;
  /* background-color: #f1f1f1; */
  background-color: rgb(255, 255, 255);
  /* margin-left: 5%;
  margin-right: 5%;  */
  margin: 2%;
  width: 20%;
  height: 300px;
  padding: 5PX;
}

.container {
  text-align: center;
  font-family: Arial, Helvetica, sans-serif;
}

.heart {
  width: 10%; 
  float: right;
}

.image {
  width: 100%;
  height: 50%;
}

#readTime {
  float: right;
}

.dateTime {
  margin-top: 2%;
  display: flex;
  justify-content: space-between;
}
