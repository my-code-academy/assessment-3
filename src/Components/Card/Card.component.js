import React, { Component } from 'react';
import './Card.component.css';
import BlackHeartImage from '../../Icons/heart-black.svg'
import RedHeartImage from '../../Icons/heart-red.svg';


class Card extends Component {
    state = {
        open: this.props.liked,
    }

    toggleImage = () => {
        this.setState(state => ({ open: !state.open }))
    }
    
    getImageName = () => this.state.open ? BlackHeartImage : RedHeartImage;

    render() {
        return (
            <div class="card">
                <img src={this.props.imageSrc} alt="abstract" className='image' />
                <div className="container">
                    <p className='title'>{this.props.title}</p>
                    <p className='abstract'>{this.props.name}</p>
                    <div className="responses">
                        <span>{this.props.rating}</span>
                        <img alt="" className="heart" src={this.getImageName()} onClick={this.toggleImage}/>
                    </div>
                </div>
            </div>
        );
    }
}

export default Card;
