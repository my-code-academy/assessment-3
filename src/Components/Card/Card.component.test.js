import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Card from './Card.component';

describe('Card', () => {
  it ('should match the snapshot', () => {
    const tree = renderer.create(<Card />);
    expect(tree).toMatchSnapshot();
  });
})