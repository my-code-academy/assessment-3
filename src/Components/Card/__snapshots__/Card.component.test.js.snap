// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
<div
  class="card"
>
  <img
    alt="abstract"
    className="image"
  />
  <div
    className="container"
  >
    <p
      className="title"
    />
    <p
      className="abstract"
    />
    <div
      className="responses"
    >
      <span />
      <img
        alt=""
        className="heart"
        onClick={[Function]}
        src="heart-red.svg"
      />
    </div>
  </div>
</div>
`;
